store/store.go

```diff
@@ -602,6 +612,76 @@ func (s *Store) doRequest(client *http.Client, reqOptions *requestOptions, user
 	return resp, err
 }
 
+
+type decodeFunc func(resp *http.Response) (error, bool)
+
+func (s *Store) retryRequest(client *http.Client, reqOptions *requestOptions, user *auth.UserState, retryStrategy retry.Strategy, decode decodeFunc) error {
```

Comment:
Yeah. I was initially skeptical and made it the way it is as to avoid repetitions in all the places where we will need retry and currently call doRequest method. But after your comment I've re-evaluated this approach and now I think that surrounding the doRequest (retryRequest) method with a loop even if a bit repetitive leads to more readable code.